pkg/service/roommanager.go

```diff
@@ -317,40 +317,48 @@ func (r *RoomManager) getOrCreateRoom(ctx context.Context, roomName livekit.Room
 		return nil, err
 	}
 
-	// construct ice servers
-	room = rtc.NewRoom(ri, *r.rtcConfig, &r.config.Audio, r.telemetry)
-	room.Hold()
+	r.lock.Lock()
 
-	r.telemetry.RoomStarted(ctx, room.Room)
+	currentRoom := r.rooms[roomName]
+	if currentRoom != lastSeenRoom {
```

Comment:
slightly more readable

```suggestion
	if currentRoom != nil {
```

Reply:
If we do that, a failure to `lastSeenRoom.Hold()` would not incur rtc room creation.

Reply:
you are right. that condition could happen.